tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/internal/types/TUint16Mapper.java

```diff
@@ -0,0 +1,136 @@
+/*
+ *  Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *  =======================================================================
+ */
+package org.tensorflow.internal.types;
+
+import org.bytedeco.javacpp.PointerScope;
+import org.tensorflow.RawTensor;
+import org.tensorflow.SparseTensor;
+import org.tensorflow.TensorMapper;
+import org.tensorflow.internal.buffer.TensorBuffers;
+import org.tensorflow.ndarray.buffer.ByteDataBuffer;
+import org.tensorflow.ndarray.impl.dense.ByteDenseNdArray;
+import org.tensorflow.ndarray.impl.sparse.ByteSparseNdArray;
+import org.tensorflow.proto.framework.DataType;
+import org.tensorflow.types.TInt64;
+import org.tensorflow.types.TUint16;
+
+/**
+ * Maps memory of {@link org.tensorflow.proto.framework.DataType#DT_Uint16} tensors to a
+ * n-dimensional data space.
+ */
+public final class TUint16Mapper extends TensorMapper<TUint16> {
+
+  @Override
+  protected TUint16 mapDense(RawTensor tensor) {
+    ByteDataBuffer buffer = TensorBuffers.toBytes(nativeHandle(tensor));
+    return new DenseTUint16(tensor, buffer);
+  }
+
+  @Override
+  protected SparseTensor<TUint16> mapSparse(
+      TInt64 indices, TUint16 values, TInt64 denseShape, PointerScope tensorScope) {
+    return new TUint16Mapper.SparseTUint16(indices, values, denseShape, tensorScope);
+  }
+
+  private static final class DenseTUint16 extends ByteDenseNdArray implements TUint16 {
```

Comment:
I think this should probably extend `ShortDenseNdArray` rather than `ByteDenseNdArray`, with similar replacements elsewhere in the code.

Reply:
Thanks for catching this, I've updated all byte references to shorts.